packages/trpc/server/routers/viewer/workflows/util.ts

```diff
@@ -99,16 +95,80 @@ export const verifyEmailSender = async (email: string, userId: number, teamId: n
           teamId,
         },
       });
-    } else if (verifiedEmail.teamId !== teamId) {
+    } else if (teamId && verifiedEmail.teamId !== teamId) {
+      await prisma.verifiedEmail.create({
+        data: {
+          email,
+          userId,
+          teamId,
+        },
+      });
+    }
+    return;
+  }
+
+  const userEmail = await prisma.user.findFirst({
+    where: {
+      id: userId,
+      email,
+    },
+  });
+
+  if (userEmail) {
+    await prisma.verifiedEmail.create({
+      data: {
+        email,
+        userId,
+        teamId,
+      },
+    });
+    return;
+  }
+
+  if (teamId) {
+    const team = await prisma.team.findFirst({
+      where: {
+        id: teamId,
+      },
+      select: {
+        members: {
+          include: {
+            user: {
+              select: {
+                id: true,
+                email: true,
+              },
+            },
+          },
+        },
+      },
+    });
```

Comment:
Is it possible to specify some other teamId that the user doesn't have control over and do something fishy with it later in the flow?

Reply:
we check here if the user is a member of the team: https://github.com/calcom/cal.com/pull/15965/files#diff-f6ef6880041c2afa0023a8b231030af6fc89b5c067f3920efa906890c100734fR154

The userId is the loggedInUser

Reply:
Oh, I missed it 😅 I was thinking of a where clause here.